src/sparseml/pytorch/optim/modifier_quantization.py

```diff
@@ -428,6 +468,61 @@ def _enable_module_qat(self, module: Module):
         if self._quantize_embeddings:
             prepare_embeddings_qat(module, reduce_range=self._reduce_range)
         self._qat_enabled = True
+        self._calibrate_if_possible(module)
+
+    def _calibrate_if_possible(self, module):
+        if self.num_calibration_steps == 0 and self._calibration_dataloader:
+            warnings.warn(
+                f"num_calibration_steps is {self.num_calibration_steps}."
+                f"Calibration data loader will not be used."
+            )
+        elif self.num_calibration_steps and not self._calibration_dataloader:
+            raise ValueError(
+                f"num_calibration_steps is {self.num_calibration_steps}. "
+                "Calibration data loader is not set."
+            )
+
+        elif not self._calibration_dataloader or not self._qat_enabled:
+            return
+
+        elif self._calibration_dataloader:
+            self._calibrate(module)
+
+    def _calibrate(self, module):
+        _LOGGER.info("Running quantization calibration using calibration_dataloader")
+
+        module_training = module.training
+        module.eval()
+
+        forward_fn: Callable = (
+            self._calibration_function
+            if self._calibration_function
+            else tensors_module_forward
+        )
+
+        model_device = next(module.parameters()).device
+        _dataloader = (
+            self._calibration_dataloader
+            if self.num_calibration_steps is None
+            else cycle(self._calibration_dataloader)
+        )
+        with suppress(StopIteration):
+            batch = next(_dataloader)
+            batch_number = 1
+            done = False
+            while not done:
+                batch = tensors_to_device(batch, model_device)
+                with torch.no_grad():
+                    forward_fn(batch, module=module)
+                done = (
```

Comment:
any reason not to `break`?

Reply:
Chose not to put a break cause it wouldn't add any extra optimizations

Reply:
the done logic is a bit spread out along with while loop. thoughts against
```python
for idx, batch in enumerate(_dataloader):
    forward
    if (...):
        break
```

Reply:
Done!